### PR TITLE
adding FromCSVToList annotation

### PR DESCRIPTION
**Pull request by tamagoko**
adding this annotation transforms the model and returns
	it as a type T which is specified in the Annotated

	documenting the usage for FromCSVToList

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -250,6 +250,35 @@ if you desire better validation for list, set, or dict fields, this must most li
...
+        list_from_string_int: FromCSVToList[List[int]]
+        # This annotation will convert the string into a list of strings
+        list_from_string: FromCSVToList[List[str]]
+        # This annotation will convert the string into a list of ints or None if the string is null or empty
+        list_from_string_int_nullable: FromCSVToList[List[int] | None]
+        # This annotation will convert the string into a list of strings or None if the string is null or empty
+        list_from_string_nullable: FromCSVToList[List[str] | None]
+
+        # if using python < 3.9, you can use typing.Union instead of the pipe operator
+        # list_from_string_nullable: FromCSVToList[Union[List[str],None]]
```
Comment (bluesliverx) [inline README.rst:279]: I think this needs to be removed now since python < 3.9 is not supported.

File: dysql/annotations.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+"""
+Copyright 2021 Adobe
```
Comment (bluesliverx) [inline dysql/annotations.py:2]: Probably should be 2023

File: dysql/annotations.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+from typing import TypeVar, Annotated
+
+from pydantic import BeforeValidator
+
+# pylint: disable=invalid-name
+T = TypeVar('T')
+
+
+# we could add validators to the base model, something like @validator('csv*', pre=True)?
```
Comment (bluesliverx) [inline dysql/annotations.py:17]: Does this note need to be removed?

File: dysql/test/test_annotations.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
+"""
+Copyright 2021 Adobe
```
Comment (bluesliverx) [inline dysql/test/test_annotations.py:2]: 2023

File: dysql/annotations.py (hunk 2 above)
Comment (tamagoko) [reply dysql/annotations.py:2]: :/

**Review by bluesliverx — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (tamagoko) [reply README.rst:279]: turns out it's still needed for 3.9, I I updated the comment

File: .pylint-license-header [hunk 5]
```diff
@@ -1,5 +1,5 @@
 """
-Copyright 2021 Adobe
+Copyright 20\d\d Adobe
```
Comment (bluesliverx) [inline .pylint-license-header:2]: I'll call this the y2.1k bug ;)

**Review by bluesliverx — APPROVED** [review]

File: .pylint-license-header (hunk 5 above)
Comment (tamagoko) [reply .pylint-license-header:2]: :P

**Pull request merged**